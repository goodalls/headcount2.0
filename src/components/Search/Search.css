.search {
  grid-column: 1/7;
  background-color: lightblue; }
  .search button {
    background-color: #F7C74A;
    color: #273B7A;
    font-family: "Lato", sans-serif;
    height: 36px;
    font-size: 1.2em;
    width: 100px;
    border-radius: 10px; }
  .search input {
    font-family: "Lato", sans-serif;
    font-size: 1.2em;
    padding: 8px;
    border-radius: 10px;
    border: none;
    width: 400px;
    height: 40px;
    margin: 35px 0 15px; }

.search-form {
  display: flex;
  flex-direction: column;
  align-items: center; }
