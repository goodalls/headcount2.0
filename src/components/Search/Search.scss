@import '../../variables.scss';

.search {
  grid-column: 1/7;
  background-color: lightblue;
  height: 100px;
  margin: 0 2%;
}