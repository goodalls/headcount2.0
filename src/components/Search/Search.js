import React, { Component } from 'react';
import './Search.css';
import PropTypes from 'prop-types';

export class Search extends Component {
  constructor(props) {
    super(props);

    this.state = {
      location: ''
    };
  }

  handleInputChange = event => {
    const { value } = event.target;

    this.setState({ location: value });
    this.props.handleSearch(value.toUpperCase());
  };

  handleSubmit = event => {
    event.preventDefault();
  };

  render() {
    return (
      <div className="search">
        <form onSubmit={this.handleSubmit}>
          <input
            onChange={this.handleInputChange}
            value={this.state.location}
            type="text"
            placeholder="Search"
          />
          <button onClick={() => this.props.handleSearch()}>Show All</button>
        </form>
      </div>
    );
  }
}

Search.propTypes = {
  handleSearch: PropTypes.func
};

export default Search;
