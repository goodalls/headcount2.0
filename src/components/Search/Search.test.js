import React from 'react';
import Search from './Search';
import { shallow } from 'enzyme';

describe('Search', () => {
  let renderedComponent;

  beforeEach(() => {
    renderedComponent = shallow(<Search />);
  });

  it('should match the snapshot', () => {
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should start with a default location state of an empty string', () => {
    expect(renderedComponent.state()).toEqual({ location: '' });
  });

  it('should update state and call method passed down from props', () => {
    /* eslint-disable */
    const mockSubmit = jest.fn();
    /* eslint-enable */
    const renderedComponent = shallow(
      <Search handleDistrictSearch={mockSubmit} />
    );

    renderedComponent.find('input').simulate('change', {
      target: {
        value: 'a'
      }
    });

    expect(renderedComponent.state().location).toEqual('a');
    expect(mockSubmit.mock.calls.length).toBe(1);
  });
});
