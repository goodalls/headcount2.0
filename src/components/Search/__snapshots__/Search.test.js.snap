// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Search />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <form
          className="search-form"
          onSubmit={[Function]}
>
          <input
                    onChange={[Function]}
                    placeholder="Search districts"
                    type="text"
                    value=""
          />
</form>,
        <button
          onClick={[Function]}
>
          Show All Districts
</button>,
      ],
      "className": "search",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <input
            onChange={[Function]}
            placeholder="Search districts"
            type="text"
            value=""
/>,
          "className": "search-form",
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "onChange": [Function],
            "placeholder": "Search districts",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        "type": "form",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Show All Districts",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Show All Districts",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <form
            className="search-form"
            onSubmit={[Function]}
>
            <input
                        onChange={[Function]}
                        placeholder="Search districts"
                        type="text"
                        value=""
            />
</form>,
          <button
            onClick={[Function]}
>
            Show All Districts
</button>,
        ],
        "className": "search",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <input
              onChange={[Function]}
              placeholder="Search districts"
              type="text"
              value=""
/>,
            "className": "search-form",
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "onChange": [Function],
              "placeholder": "Search districts",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          "type": "form",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Show All Districts",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Show All Districts",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
